### ISSUE TITLE
Uglification error after moving from 2.2.1 to 2.3.0

### ISSUE THREAD

**Issue by nolde** [RasCarlito/axios-cache-adapter#78]
My application build started failing after moving from 2.2.1 to 2.3.0:

```
ERROR in assets/js/lib.17dbc7ed.js from UglifyJs
Unexpected token: name (CacheControl) [assets/js/lib.17dbc7ed.js:45088,6]
```

Moving back to 2.2.1 fixes the issue. Seems related to `@tusbar/cache-control`.

Using `webpack@3.10.0` and the included uglify plugin.

**Comment by RasCarlito** [comment]
Hey @nolde !

Thanks for reporting this. Will look into it ASAP.

Cheers

**Comment by RasCarlito** [comment]
I don't have the issue when using `webpack@4` to build an app which is using `axios-cache-adapter@2.3.0`. Don't have time to try it out with `webpack@3` right now.

**Comment by nolde** [comment]
Thanks for the quick response, @RasCarlito !

I am also having a hard time pinpointing the exact issue. I'll have to create a project to simulate only this issue.

Meanwhile, I did notice that you do not generate an ES6 module output for you library. Maybe adding that would allow for more control over the process from my side, as I would have `@tusbar/cache-control` as part of my node_modules.

I will update this when if I find anything new. For now, moving back to 2.2.1 is fine.

**Comment by RasCarlito** [comment]
There is probably an issue with my webpack build of the `dist` folder.

Good point about exposing an ES6 module, I'll add an issue to tackle that subject. 😄 

I did not want an extra external dependency so `@tusbar/cache-control` is being bundled into the dist version on the contrary of `lodash` and `axios` which are kept external.
But maybe that is the issue and I'll just stop bundling `@tusbar/cache-control` 🤔 

If you have time extracting the issue into a seperate project that would be of great help!

**Comment by nolde** [comment]
Hi @RasCarlito !

I have created a repo that can reproduce the issue: https://github.com/nolde/axios-cache-test

That said, the reason is that `class CacheControl` has not been transformed by babel. Uglify does not understand it and breaks. I would say that, given that you are already converting everything else in the package, making sure ES6 classes are converted as well seems to make sense, and would fix the issue.

Meanwhile, I'll try to tell babel, on my side, to handle that.

**Comment by RasCarlito** [comment]
Ok! Awesome thanks.

So it does not fail for me because I don't have Uglify configured in my test and I run the code in latest chrome so classes are supported.

Cheers

**Comment by nickverstocken** [comment]
@nolde have you found a workaround for this? Having the same issue but haven't found a way to solve it...

**Comment by nolde** [comment]
Hi @nickverstocken!

I tried to make babel parse it in my side, but I have been unable to do it reliably. Didn't try too hard, though. :-)

My workaround is to move back to version 2.2.1, which does not have the same problem. If you do not have a hard dependency on any new features, I would recommend this option until @RasCarlito fixes the transpilation issue on his side.

**Comment by nickverstocken** [comment]
@nolde  Yeah I did the same the only downside now is that in my webapp I can't use the `ignoreCache` option anymore. I hope this get fixed soon or I will take a look to create PR on my own 😉 Thanks anyway for thde reply!

**Comment by RasCarlito** [comment]
Tried to fix this in #79 but no luck so far 😢

**Comment by nolde** [comment]
Could you test my change to verify whether it works, @RasCarlito ? I could not build it in Windows.

**Comment by RasCarlito** [comment]
Awesome! Will try it out ASAP ;)

**Comment by nolde** [comment]
Hello @RasCarlito !

Just FYI, I just tried to build it using a Linux image through docker. I thought the problem was Windows, but since I got the _exact same error_, I would say that there is something wrong in your setup, but for some reason your current environment is masking it. 

Maybe your using some tool installed globally?

The error is the following:
```
/home/axios-cache-adapter # npm run build

> axios-cache-adapter@2.3.0 build /home/axios-cache-adapter
> npm run clean && webpack && NODE_ENV=production webpack && NODE_BUILD_FOR=node webpack && NODE_ENV=production NODE_BUILD_FOR=node webpack


> axios-cache-adapter@2.3.0 clean /home/axios-cache-adapter
> rimraf dist

/home/axios-cache-adapter/node_modules/webpack-cli/bin/config-yargs.js:89
                                describe: optionsSchema.definitions.output.properties.path.description,
                                                                           ^

TypeError: Cannot read property 'properties' of undefined
    at module.exports (/home/axios-cache-adapter/node_modules/webpack-cli/bin/config-yargs.js:89:48)
    at /home/axios-cache-adapter/node_modules/webpack-cli/bin/webpack.js:60:27
    at Object.<anonymous> (/home/axios-cache-adapter/node_modules/webpack-cli/bin/webpack.js:515:3)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
```

**Comment by nolde** [comment]
@RasCarlito , did you ever get around testing my thing?

**Comment by miluoshi** [comment]
Shouldn't the bundled version in `dist` folder be completely transpiled to ES5? Now it contains untranspiled `class CacheControl` which breaks IE11 with syntax error.

**Comment by RasCarlito** [comment]
Hey @nolde just left you a comment on #80 😄

**Comment by nolde** [comment]
Hi @RasCarlito ! 

Awesome! Just left you a message on #80 as well!

**Comment by nolde** [comment]
Thanks for your help, @RasCarlito !

**Comment by nolde** [comment]
Hi @RasCarlito !

Sorry to be obnoxious, but do you have any idea on when you'll be able to release 2.3.1 with this fix in it?

**Comment by RasCarlito** [comment]
Hi @nolde ! No worries sorry it takes so long.

I was about to release it but when trying out the build one last time it did not work. The `cache-control` code was left as a `class` in the dist version 😢 

So I started working on another solution here #91 but it lead me to an issue when transpiling the `cache-control` code which uses `module.exports` while the rest of my code base uses `export default`.

I forked `cache-control` to change the export syntax and pointed my dependency directly on my fork without publishing it to npm (I think it's ok as I only want to bundle the dependency in the dist version). But now my travis builds are unhappy about installing my forked version of `cache-control` from github directly 😞 

I am re-opening this issue in the meantime. I really want to fix this quickly.

I could consider using the solution in #90 which finally leaves the transpilation burden to the user land.

Else I need to publish my forked version of `cache-control` so that the travis builds pass.

Again, sorry it takes so long.

Thanks for everyones inputs and help on this 🚀 

Cheers 🍻

**Comment by RasCarlito** [comment]
Should now be fixed with #91 will at last make the 2.3.1 release 🎉

**Comment by RasCarlito** [comment]
Released >> [2.3.1](https://github.com/RasCarlito/axios-cache-adapter/releases/tag/2.3.1)

@nolde Could you confirm it fixes your initial issue?

Cheers 🍻

**Comment by nolde** [comment]
Issue solved, just tested using [v2.3.2](https://github.com/RasCarlito/axios-cache-adapter/releases/tag/2.3.2)!

Thanks @RasCarlito !
